pyEpiabm/pyEpiabm/core/_compartment_counter.py

```diff
@@ -47,9 +53,17 @@ def report(self, old_status: InfectionStatus,
         :param new_status: Person's new infection state
         :type new_status: InfectionStatus
         """
+        if self._compartments[old_status] == 0:
```

Comment:
```suggestion
        if self._compartments[old_status] <= 0:
```

Should never be negative anyway (in which case it will be equivalent to the old version) but just in case?